src/poisson_solvers.jl

```diff
@@ -1,17 +1,21 @@
 import FFTW
 import GPUifyLoops: @launch, @loop, @synchronize
 
-function init_poisson_solver(::CPU, g::Grid, tmp_rhs)
-    tmp_rhs.data .= rand(Float64, g.Nx, g.Ny, g.Nz)
-    PoissonSolver(g, tmp_rhs, FFTW.MEASURE)
-end
-
-function init_poisson_solver(::GPU, g::Grid, tmp_rhs)
-    tmp_rhs.data .= CuArray{Complex{Float64}}(rand(Float64, g.Nx, g.Ny, g.Nz))
-    PoissonSolverGPU(g, tmp_rhs)
+PoissonSolver(::CPU, grid::Grid) = PoissonSolverCPU(grid)
+PoissonSolver(::GPU, grid::Grid) = PoissonSolverGPU(grid)
+
+struct PoissonSolverCPU{A1<:AbstractArray, A3<:AbstractArray} <: PoissonSolver
+    kx²::A1
+    ky²::A1
+    kz²::A1
+    storage::A3
+    FFT!
```

Comment:
perhaps make the types of the plans concrete? just need to give them names --- julia will figure out the rest.

Reply:
Good point. Will do.

Reply:
Done.